crates/sncast/src/helpers/scarb_utils.rs

```diff
@@ -154,7 +154,10 @@ pub fn build_and_load_artifacts(
     build(package, config).map_err(|e| anyhow!(format!("Failed to build using scarb; {e}")))?;
 
     let metadata = get_scarb_metadata_with_deps(&config.scarb_toml_path)?;
-    get_contracts_map(&metadata, &package.id, Some(&config.profile))
+    match &metadata.profiles.contains(&config.profile) {
```

Comment:
How about displaying a warning message regarding the Scarb profile being used in the following scenarios?
1. No profile has been provided 
2. The provided profile does not exist

Reply:
I think a message informing what scarb profile is used is a good thing in general

Reply:
Yeah, makes sense, I'd do it only for profiles that do not exist in scarb (as I don't think we should inform when profile is not set, scarb does not inform about its default and neither should we imo).

Something like: "Profile <name> does not exist in scarb, using the default "dev" profile" ?

Reply:
Sounds good

Reply:
added warn message in https://github.com/foundry-rs/starknet-foundry/pull/1695/commits/446fbb3b942438cfbcfe4c972ba27074a85c5905